src/controllers/hid/hidcontroller.cpp

```diff
@@ -298,6 +298,34 @@ void HidController::sendBytesReport(QByteArray data, unsigned int reportID) {
     }
 }
 
+void HidController::sendFeatureReport(
+        const QList<int>& dataList, unsigned int reportID) {
+    QByteArray dataArray;
+    for (const int datum: dataList) {
+        dataArray.append(datum);
+    }
+	
+    // Append the Report ID to the beginning of dataArray[] per the API..
+    dataArray.prepend(reportID);
+
+    int result = hid_send_feature_report(
+            m_pHidDevice, (unsigned char*)dataArray.constData(), dataArray.size());
+    if (result == -1) {
+        if (ControllerDebug::enabled()) {
+            qWarning() << "Unable to send data to" << getName()
+                       << "serial #" << hid_serial << ":"
```

Comment:
Only removing the serial number in non-debug mode is not significant enough to justify having these two branches.  Either always send a message, or only send a message in debug mode.  I'd prefer we only send this message in debug mode.  It will probably only be useful during development or if the controller isn't working, and in both cases the user will want to run with debugging enabled in order to diagnose the problem.

Reply:
On further thought, I think a warning of a script failure should always be shown regardless of ControllerDebug::enabled

Reply:
I used excatly the code as for HidController::sendBytesReport 20 lines above. Do you prefer two different implementations for the same type of wrapper code?

Reply:
Warning message is now always print, independent of ControllerDebug::enabled()